### PR TITLE
Explorer: Fix domain table and bump @bonfida/spl-name-service to 0.1.30

### PR DESCRIPTION
**Pull request by aletna**
#### Problem
#24271 build failed due to breaking changes in the package's new versions.

#### Summary of Changes
The `NameRegistryState` class does not have an `address` field in versions `0.1.23` & above and `retrieveBatch()` does not return the address either so I rewrote `performReverseLookup()` to account for that. The change doesn't break the domain display and load time seems similar, though I haven't tested it with accounts with a large amount of domains.

What do you think @jstarry ?

_**Note:**_ for some reason I could only get it to run by installing bootstrap's peer dependency `@popperjs/core`. Not sure if it's a needed change.

Fixes #

### CODE REVIEW

File: explorer/src/utils/name-service.tsx [hunk 1]
```diff
@@ -55,6 +55,30 @@ export async function findOwnedNameAccountsForUser(
...
+
+  const nameLength = new BN(nameRegistry!.data.slice(0, 4), "le").toNumber();
+  return {
+    name: nameRegistry.data.slice(4, 4 + nameLength).toString() + ".sol",
+    address: domainKey,
+    class: nameRegistry.class,
+  };
+}
+
 export async function performReverseLookup(
```
Comment (jstarry) [inline explorer/src/utils/name-service.tsx:82]: Can you migrate to using the `performReverseLookup` function provided by the name service package? https://github.com/Bonfida/solana-program-library/blob/dc0b43d690087d1fb12873d1f3c8218bee8e368a/name-service/js/src/utils.ts#L170

**Comment by jstarry** [comment]
> Note: for some reason I could only get it to run by installing bootstrap's peer dependency @popperjs/core. Not sure if it's a needed change.

Shouldn't be needed, what issue are you running into?

**Comment by aletna** [comment]
> Shouldn't be needed, what issue are you running into?

not quite sure why but been getting this error after installing all packages and trying to run the explorer locally. Something about a missing bootstrap peer dependency. Every time i ran npm install it removed all popperjs peer dependency entries from the package-lock file. 

```
./node_modules/bootstrap/dist/js/bootstrap.min.js
Module not found: Can't resolve '@popperjs/core' in '~/solana/explorer/node_modules/bootstrap/dist/js'
```

just added some changes and removed @core/popperjs before pushing it to this branch. Since the remote build passed, it was likely just a local issue. The update @bonfida/spl-name-service logic was not affected by it.

**Comment by jstarry** [comment]
During review, I got really confused what the original implementation was trying to do. It was basically wrong so I rewrote it and incorporated your changes which are now in a pull request to your branch: https://github.com/aletna/solana/pull/1

Do you mind looking it over to see it makes sense? If so, feel free to merge into your branch and then we can get this PR merged into master.

**Comment by aletna** [comment]
Thanks for diving into this - looks great! 

Agreed, the original implementation of `class` doesn't make much sense anymore and variable names are more understandable now as well (esp `userDomains` instead of `names`). 

Included it in the branch!

**Comment by jstarry** [comment]
Sweet! In case it wasn't clear, the live implementation shows the domain and class addresses for the reverse lookup registry instead of the domain account for the .sol domain registry.

**Pull request merged**